Applications/FileIO/GocadIO/GocadAsciiReader.cpp

```diff
@@ -142,35 +191,38 @@ bool GocadTSurfaceReader::TSurfaceFound(std::ifstream& in) const
         {
             continue;
         }
-        if (line.substr(0, 11) == "GOCAD TSurf")
+
+        if (line.substr(0, 10) == "GOCAD VSet")
         {
-            return true;
-        // No idea why this is allowed in a *.ts file.
-        // It should be a whole different file type.
+            return GocadDataType::VSET;
         }
-        if (line.substr(0, 13) == "GOCAD Model3d")
+        else if (line.substr(0, 11) == "GOCAD PLine")
```

Comment:
I'd move this string to enum conversion to the place where the enum is defined.

Also no else after return is needed.

Reply:
Do you suggest that I should implement a enum -> string conversion for this one use?

The return does not follow "else" but "while" and is required in case the end of the file is reached.